tests/installation/test_executor.py

```diff
@@ -737,6 +793,111 @@ def test_executor_should_write_pep610_url_references_for_urls(
         "url": package.source_url,
     }
     verify_installed_distribution(tmp_venv, package, expected_url_reference)
+    if is_artifact_cached:
+        download_spy.assert_not_called()
+    else:
+        download_spy.assert_called_once_with(
+            mocker.ANY, operation, Link(package.source_url)
+        )
+
+
+@pytest.mark.parametrize(
+    (
+        "is_sdist_artifact_cached",
+        "is_wheel_cached",
+        "expect_artifact_building",
+        "expect_artifact_download",
+    ),
+    [
+        (True, False, True, False),
+        (True, True, False, False),
+        (False, False, True, True),
+        (False, True, False, True),
+    ],
+)
+def test_executor_should_write_pep610_url_references_for_non_wheel_urls(
+    tmp_venv: VirtualEnv,
+    pool: RepositoryPool,
+    config: Config,
+    io: BufferedIO,
+    mock_file_downloads: None,
+    mocker: MockerFixture,
+    fixture_dir: FixtureDirGetter,
+    is_sdist_artifact_cached: bool,
```

Comment:
```suggestion
    is_sdist_cached: bool,
```

consistent to `is_wheel_cached`

Reply:
Fixed